src/Core/src/Hosting/Semantics/AppHostBuilderExtensions.cs

```diff
@@ -0,0 +1,17 @@
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Maui.Hosting
+{
+	public static partial class AppHostBuilderExtensions
+	{
+		public static IAppHostBuilder ConfigureSemantics(this IAppHostBuilder builder)
+		{
+			builder.ConfigureServices(services =>
+			{
+				services.AddTransient<ISemanticService>(svcs => new SemanticService());
```

Comment:
Can we make this AddSingleton? I don't think we need to return unique instances with every call to this service currently